### PR TITLE
fix: document getting first or last node exception

### PR DESCRIPTION
**Pull request by Linij**
fix issue https://github.com/AppFlowy-IO/appflowy-editor/issues/884

### CODE REVIEW

File: lib/src/core/document/document.dart [hunk 1]
```diff
@@ -75,10 +75,14 @@ class Document {
   final Node root;
 
   /// First node of the document.
-  Node? get first => root.children.first;
+  Node? get first => root.children.isNotEmpty ? root.children.first : null;
```
Comment (LucasXu0) [inline lib/src/core/document/document.dart:78]:
```suggestion
  Node? get first => root.children.firstOrNull;
```

File: lib/src/core/document/document.dart [hunk 2]
```diff
@@ -75,10 +75,14 @@ class Document {
...
 
   /// Last node of the document.
   Node? get last {
+    if (root.children.isEmpty) {
```
Comment (LucasXu0) [inline lib/src/core/document/document.dart:82]:
```dart
  Node? get last {
    Node? current = root.children.lastOrNull;
    while (current != null && current.children.isNotEmpty) {
      current = current.children.last;
    }
    return current;
  }
```

**Comment by LucasXu0** [comment]
@Linij Thanks for your contribution. Please update your comment.

**Comment by Linij** [comment]
hi,i accept the suggesstion code, and update the commit.

**Pull request merged**